packages/trader/src/Modules/Trading/Components/Form/ContractType/ContractTypeMenu/contract-type-menu.jsx

```diff
@@ -0,0 +1,219 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { CSSTransition } from 'react-transition-group';
+import { VerticalTab, ThemedScrollbars } from '@deriv/components';
+import { localize } from '@deriv/translations';
+import SearchInput from './search-input.jsx';
+import NoResultsMessage from './no-results-message.jsx';
+import { Header } from '../ContractTypeInfo';
+import { getContractCategoryLabel, getContractsList, getFilteredList } from '../../../../Helpers/contract-type';
+
+class Dialog extends React.PureComponent {
+    dialog_ref = React.createRef();
+    scrollbar_ref = React.createRef();
+    vertical_tab_headers = [];
+    is_user_scroll = false;
+
+    state = {
+        is_filtered_list_empty: false,
+        selected: this.contract_category,
+        value: this.props.item.value,
+        input_value: '',
+    };
+
+    static getDerivedStateFromProps(props, state) {
+        if (props.item.value !== state.value) {
+            return {
+                selected: {
+                    label: getContractCategoryLabel(props.list, props.item),
+                    value: props.item.value,
+                },
+            };
+        }
+        if (!props.is_open && !props.is_info_dialog_open && state.selected !== null) {
+            return {
+                selected: null, // reset selected header when dialog is closed
+            };
+        }
+        return null;
+    }
+
+    componentDidUpdate() {
+        if (this.props.is_open) {
+            if (!this.vertical_tab_headers.length) {
+                this.vertical_tab_headers = this.getVerticalTabHeaders();
+            }
+            if (this.should_scroll) {
+                this.scroll();
+            }
+        }
+    }
+
+    onChange = e => {
+        if (this.props.is_info_dialog_open) {
+            this.props.onBackButtonClick();
+        }
+        this.setState(
+            {
+                selected: e,
+            },
+            () => {
+                this.scroll();
```

Comment:
need to pass `e` to this.scroll here or else it will be undefined in the `onScroll`?

Reply:
@oskar-binary no, we don't need user event `e` here cos this function triggers scroll on the list to the position of selected contract category header